File: angularclient/src/app/model/Robots/Robot.ts
import {RobotModel} from "./RobotModel";
import {Pose} from "../Stand/Pose";
import {RobotStatus} from "./RobotStatus";
import {RobotBattery} from "./RobotBattery";
import {ExtraRobotElement} from "./ExtraRobotElement";

export class Robot{
  id: string;
  robotIp: string;
  available: boolean;
  extraRobotElement: ExtraRobotElement;
  model: RobotModel;
  battery: RobotBattery;
  pose: Pose;
  batteryLevel: number;
  status: RobotStatus[];
  timestamp: string;

  constructor(robotIp: string, available: boolean, extraRobotElement: ExtraRobotElement, model: RobotModel, battery: RobotBattery, pose: Pose, batteryLevel: number, status: RobotStatus[], timestamp: string) {
    this.robotIp = robotIp;
    this.available = available;
    this.extraRobotElement = extraRobotElement;
    this.model = model;
    this.battery = battery;
    this.pose = pose;
    this.batteryLevel = batteryLevel;
    this.status = status;
    this.timestamp = timestamp;
  }
}

File: ra-back/src/main/java/pl/raptors/raptorsRobotsApp/domain/robots/Robot.java
package pl.raptors.raptorsRobotsApp.domain.robots;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import pl.raptors.raptorsRobotsApp.domain.movement.Pose;

import java.util.List;

@NoArgsConstructor
@Data
@Document(collection = "robots")
public class Robot {

    @Id
    private String id;
    private String robotIp;
    private Boolean available;
    private ExtraRobotElement extraRobotElement;
    private RobotModel model;
    private RobotBattery battery;
    private Pose pose;
    private Double batteryLevel;
    private List<RobotStatus> status;
    private String timestamp;

    public Robot(String robotIp, Boolean available, ExtraRobotElement extraRobotElement, RobotModel model, Pose pose, String timestamp, Double batteryLevel, List<RobotStatus> status) {
        this.robotIp = robotIp;
        this.available = available;
        this.extraRobotElement = extraRobotElement;
        this.model = model;
        this.pose = pose;
        this.timestamp=timestamp;
        this.batteryLevel = batteryLevel;
        this.status = status;
    }
}
